Base/QTGUI/qSlicerModuleFinderDialog.cxx

```diff
@@ -0,0 +1,342 @@
+/*==============================================================================
+
+  Program: 3D Slicer
+
+  Copyright (c) Kitware Inc.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Julien Finet, Kitware Inc.
+  and was partially funded by NIH grant 3P41RR013218-12S1
+
+==============================================================================*/
+
+// STD includes
+#include <algorithm>
+
+// Qt includes
+#include <QKeyEvent>
+#include <QPushButton>
+
+// QtGUI includes
+#include "qSlicerAbstractCoreModule.h"
+#include "qSlicerAbstractModule.h"
+#include "qSlicerApplication.h"
+#include "qSlicerModuleFactoryFilterModel.h"
+#include "qSlicerModuleFactoryManager.h"
+#include "qSlicerModuleManager.h"
+//#include "qSlicerModulesMenu.h"
+#include "qSlicerModuleFinderDialog.h"
+#include "qSlicerUtils.h"
+#include "ui_qSlicerModuleFinderDialog.h"
+
+// --------------------------------------------------------------------------
+// qSlicerModuleFinderDialogPrivate
+
+//-----------------------------------------------------------------------------
+class qSlicerModuleFinderDialogPrivate: public Ui_qSlicerModuleFinderDialog
+{
+  Q_DECLARE_PUBLIC(qSlicerModuleFinderDialog);
+protected:
+  qSlicerModuleFinderDialog* const q_ptr;
+
+public:
+  qSlicerModuleFinderDialogPrivate(qSlicerModuleFinderDialog& object);
+
+  /// Convenient method regrouping all initialization code
+  void init();
+
+  QString CurrentModuleName;
+};
+
+// --------------------------------------------------------------------------
+// qSlicerModuleFinderDialogPrivate methods
+
+// --------------------------------------------------------------------------
+qSlicerModuleFinderDialogPrivate::qSlicerModuleFinderDialogPrivate(qSlicerModuleFinderDialog& object)
+  :q_ptr(&object)
+{
+  //this->ModulesMenu = nullptr;
+}
+
+// --------------------------------------------------------------------------
+void qSlicerModuleFinderDialogPrivate::init()
+{
+  Q_Q(qSlicerModuleFinderDialog);
+
+  this->setupUi(q);
+
+  qSlicerCoreApplication * coreApp = qSlicerCoreApplication::application();
+  qSlicerAbstractModuleFactoryManager* factoryManager = coreApp->moduleManager()->factoryManager();
+
+  qSlicerModuleFactoryFilterModel* filterModel = this->SlicerModulesListView->filterModel();
+
+  // Hide modules that do not have GUI (use cannot switch to them)
```

Comment:
`use cannot switch to them` -> `user cannot switch to them`

Reply:
ok